examples/ninja_simple/code/build.ninja

```diff
@@ -0,0 +1,43 @@
+# build.ninja
+
+ninja_required_version = 1.3
+
+# Variables
+
+builddir = rules_foreign_cc_build
+rm       = rm -fr
+cxx      = clang
```

Comment:
I don't know ninja build syntax but can this be read from the environment to pick up `CC` and `CPP_FLAGS` so that it honours the configured cc_toolchain?

Reply:
It could be but are those flags actually being set?
```
{"XCODE_VERSION_OVERRIDE": "12.4.0.12D4e", "APPLE_SDK_VERSION_OVERRIDE": "11.1", "APPLE_SDK_PLATFORM": "MacOSX"}
```
this is my `cc_env` for that action and MacOS is the only platform that doesn't use `use_default_shell_env`.

In general though, you can use an environment variable within a command by prefixing it with `$$`. I'm happy to clean up the cc_toolchain in another PR if possible and as an isolated change. It's something that should be done for the repo to ensure the rules work exactly as they're expected (and no longer be limited to what's installed on the BuildKite hosts).

Reply:
Ok, lets fix that up in another PR - I think its common for build tools to expect the c compiler to be specified in `CC`.